### PR TITLE
Add shuffle function in view learning

### LINKED ISSUE #1051: UX: design new learning screen
learning_screen

### PR DESCRIPTION
**Pull request by ksheremet**
Fixes #1051

### CODE REVIEW

File: flutter/lib/view_models/cards_view_learning_bloc.dart [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
 class CardsViewLearningBloc extends ScreenBloc {
   final ListAccessor<CardModel> _cardList;
 
   CardsViewLearningBloc({@required User user, @required DeckModel deck})
       : _cardList = deck.cards,
-        super(user);
+        super(user) {
+    _doGetCardsListController.add(_cardList.value);
+    _onShuffleStreamController.stream.listen((_) {
+      final shuffledCardList = _cardList.value.rebuild((cardListBuilder) {
```
Comment (dotdoom) [inline flutter/lib/view_models/cards_view_learning_bloc.dart]: What happens if `.value` is null?

File: flutter/lib/view_models/cards_view_learning_bloc.dart [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
 
 class CardsViewLearningBloc extends ScreenBloc {
   final ListAccessor<CardModel> _cardList;
 
   CardsViewLearningBloc({@required User user, @required DeckModel deck})
       : _cardList = deck.cards,
-        super(user);
+        super(user) {
+    _doGetCardsListController.add(_cardList.value);
+    _onShuffleStreamController.stream.listen((_) {
```
Comment (dotdoom) [inline flutter/lib/view_models/cards_view_learning_bloc.dart]: What if the source cards list updates?

File: flutter/lib/view_models/cards_view_learning_bloc.dart (hunk 1 above)
Comment (ksheremet) [reply flutter/lib/view_models/cards_view_learning_bloc.dart]: This situation shouldn't happen. If there are no cards in a deck, the user gets the 'add cards' screen when presses on it.

Anyway, I can show a dialog that there are no cards to shuffle. What do you think?

File: flutter/lib/view_models/cards_view_learning_bloc.dart (hunk 2 above)
Comment (ksheremet) [reply flutter/lib/view_models/cards_view_learning_bloc.dart]: Nothing will happen. Do you think it is important to interrupt user if new cards arrived during the learning session? User can learn them when opens deck again.

File: flutter/lib/view_models/cards_view_learning_bloc.dart (hunk 1 above)
Comment (dotdoom) [reply flutter/lib/view_models/cards_view_learning_bloc.dart]: Null would usually mean that cards aren't loaded yet

Comment (ksheremet) [reply flutter/lib/view_models/cards_view_learning_bloc.dart]: I see. I'll disable the shuffle button then

**Pull request merged**